src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -94,10 +94,23 @@ public static AppiumDriverLocalService buildService(AppiumServiceBuilder builder
     }
 
     public AppiumDriverLocalService withBasePath(String basePath) {
-        this.basePath = basePath;
+        this.basePath = sanitizeBasePath(basePath);
         return this;
     }
 
+    @SneakyThrows private static String sanitizeBasePath(String basePath) {
+        if (null == basePath) {
```

Comment:
Consider using checkNotNull helper from guava lib. An exception must be thrown if this argument is set to null

Reply:
I believe the default value for basePath is null in case the user doesn't provide a basepath at all.. I referred to the getUrl method in which its checking for null. we should not throw an exception in this case (in my opinion) rather a warning message should be good enough and appium can start with no additional basepath

Reply:
But yeah, we can highlight to user by throwing an exception that they have provided a null value in basepath, so that they can make an informed decision to either provide a valid value or no value at all. I will make the change, thanks.

Reply:
the only place this API is called is `withBasePath` builder method. And there basePath must not be null. In general we are quite careful about explicitly returning null from any method and try to avoid that as much as possible.

Reply:
In general I'd add something like `checkArgument(isBlank(basePath), "basePath must not be empty or null")`

Reply:
Fixed this by moving the validations to "withArgument" method and checking for null, blank and empty values. Introduced a InvalidBasePathException to throw in case of any validations fail.